### ISSUE TITLE
False positive: Valid RDS instance types reported as invalid since v1.47.0

### ISSUE THREAD

**Issue by Tommes-M** [aws-cloudformation/cfn-lint#4445]
### CloudFormation Lint Version

v1.48.1

### What operating system are you using?

Windows

### Describe the bug

When validating CloudFormation templates that use certain RDS instance types (e.g. `db.t4g.2xlarge`), cfn-lint reports an error like:

```
'db.t4g.2xlarge' is not one of ['db.m4.large', 'db.m4.xlarge', ...]
```

This is a false positive. These instance types are valid and actively in use — our RDS instances are running on them without any issues.

### Expected behavior

`db.t4g.2xlarge` (and other valid instance types) should be accepted without errors.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  MyDB:
    Type: AWS::RDS::DBInstance
    Properties:
      DBInstanceClass: db.t4g.2xlarge
      Engine: mysql
      MasterUsername: admin
      MasterUserPassword: !Sub "{{resolve:secretsmanager:my-secret}}"
      AllocatedStorage: "20"
```
